department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-4125- Reassign Issues to a Split Appeal

Resolves [APPEALS-4125](https://vajira.max.gov/browse/APPEALS-4125)

### Description
As a developer I need to assign issues to the original and new appeals so that issues are split between appeals

### Acceptance Criteria

When clicking "Split Appeal" on the Review and Confirm page of Split Appeal Workflow form:
- [ ] Issues in the form's new issue list are assigned to the new appeal.
- [ ] Issues in the form not in the new issue list are maintained on the original appeal
- [ ] Issue assignments are wrapped in the same transaction as work done in [APPEALS-4156](https://vajira.max.gov/browse/APPEALS-4156) so if any step of splitting an appeal fails all changes are rolled back

### Testing Plan
1. Go to demo login and sign in as a SSC member (SPLTAPPL members). 
2. Go to **Queue**, then **search cases**. Enter an appeal veteran file number or docket stream number. I used the veteran file number 673020006 in this case. 
![image](https://user-images.githubusercontent.com/99915461/199591372-3dad4431-6aa9-4333-9aa8-e27e38667462.png)

3. Click on the docket number to open up the case details page. 
4. Click on the **correct issues** to add issues to the appeal.
![image](https://user-images.githubusercontent.com/99915461/199591498-07c1d86a-f3ce-419d-926a-0d39e835c6af.png)

5. Click **Add Issues** to add an issue to the appeal. Fill out the form and click **continue** to add the issue. Do this at least twice to gain access to the split appeal button. When you are finished click **save** and **yes, save**.
![image](https://user-images.githubusercontent.com/99915461/199591718-ecbd0155-f393-4322-8c4d-cee79afaa39d.png)
![image](https://user-images.githubusercontent.com/99915461/199591800-de267412-9783-4a27-bc00-5df9c0fabcdc.png)

6. Click the **correct issues** link again to get back to the edit page. Click the **split appeal** button. 
7. On the Create New Appeal Stream page, fill out the form and select both issues. The **continue** button should be greyed out. Deselect one of the issues and  Click **continue**.
![image](https://user-images.githubusercontent.com/99915461/199592054-de7a81c3-0be6-4940-ba16-04df558cbe3d.png)

8. On the last page, click **split appeal**.
9. You should be greeted by a success banner. 
![image](https://user-images.githubusercontent.com/99915461/199592173-370f4541-54c8-45d9-a9ca-6fc69364aceb.png)

10. Make sure the appeal was successfully split by going to the **search cases** link and put in the veteran file number and/or docket number from earlier. The new appeal should only have 1 issue on it. Click on the new appeal and confirm the issue was the issue you selected.
![image](https://user-images.githubusercontent.com/99915461/199592972-9f6b737d-ceb7-4da2-872b-10d632b53ed1.png)
